### PR TITLE
Update Rusqlite (0.16 -> 0.24)

### PR DESCRIPTION
**Pull request by lgalabru**
This PR is addressing https://github.com/blockstack/stacks-blockchain/issues/1274.
We're looking at 3 different refactoring:
- `row::get_unwrap` is the new `row::get`
- `query_row` now returning a `Result<T, rusqlite::Error>`
- `rows::next` now returning a `Result<Option<T>>` (previously returning `Option<Result<T>>`)

### CODE REVIEW

File: src/chainstate/stacks/db/blocks.rs [hunk 1]
```diff
@@ -914,16 +914,10 @@ impl StacksChainState {
...
-                Ok(row) => {
-                    let next_blob: Vec<u8> = row.get(0);
-                    blobs.push(next_blob);
-                }
-                Err(e) => {
-                    return Err(Error::DBError(db_error::SqliteError(e)));
-                }
-            };
+
+        while let Some(row) = rows.next().map_err(|e| db_error::SqliteError(e))? {
```
Comment (zone117x) [inline src/chainstate/stacks/db/blocks.rs:918]: Error handling in a rust destructuring loop 🥇

File: src/clarity.rs [hunk 2]
```diff
@@ -203,21 +195,16 @@ fn get_cli_block_height(conn: &Connection, block_id: &StacksBlockId) -> Option<u
...
-                );
-                row_opt = Some(rowid);
-                break;
-            }
-            Err(e) => {
-                panic!("FATAL: could not read block hash: {:?}", e);
-            }
-        }
+
+    while let Some(row) = rows.next().expect("FATAL: could not read block hash") {
```
Comment (zone117x) [inline src/clarity.rs:199]: The sqlite error message is no longer included, were they helpful?
`panic!("FATAL: could not read block hash: {:?}", e);`

**Review by zone117x — APPROVED** [review]
LGTM!

File: src/chainstate/stacks/db/blocks.rs (hunk 1 above)
Comment (lgalabru) [reply src/chainstate/stacks/db/blocks.rs:918]: yeah chaining the `map_err` is not ideal. I've been trying a few loop patterns, this one was the most readable one I could come up with - open to something else.

**Comment by jcnelson** [comment]
We'll want all the unit and integration tests to pass at least locally before merging, but otherwise this looks good to me.

File: src/chainstate/stacks/db/blocks.rs (hunk 1 above)
Comment (zone117x) [reply src/chainstate/stacks/db/blocks.rs:918]: I think for anyone familiar with the destructuring and `?` syntax it is readable enough

**Comment by lgalabru** [comment]
> We'll want all the unit and integration tests to pass at least locally before merging, but otherwise this looks good to me.

Tests are passing after a re-run!

**Review by jcnelson — APPROVED** [review]

**Pull request merged**